test/sync/src/commonMain/kotlin/io/realm/kotlin/test/mongodb/util/TestAppInitializer.kt

```diff
@@ -0,0 +1,571 @@
+                      "_id": {
+                        "bsonType": "objectId"
+                      },
+                      "name": {
+                        "bsonType": "string"
+                      },
+                      "realm_id": {
+                        "bsonType": "string"
+                      }
+                    },
+                    "required": [
+                      "name",
+                      "_id"
+                    ],
+                    "title": "FlexChildObject"
+                }
+            }
+            """.trimIndent()
+        )
+
+        app.addSchema(
+            """
+            {
+                "metadata": {
+                    "data_source": "BackingDB",
+                    "database": "$databaseName",
+                    "collection": "FlexParentObject"
+                },
+                "relationships": {
+                    "child": {
+                        "ref": "#/relationship/BackingDB/$databaseName/FlexChildObject",
+                        "source_key": "child",
+                        "foreign_key": "_id",
+                        "is_list": false
+                    }
+                },
+                "schema": {
+                    "properties": {
+                        "_id": {
+                          "bsonType": "objectId"
+                        },
+                        "age": {
+                          "bsonType": "int"
+                        },
+                        "child": {
+                          "bsonType": "objectId"
+                        },
+                        "name": {
+                          "bsonType": "string"
+                        },
+                        "realm_id": {
+                          "bsonType": "string"
+                        },
+                        "section": {
+                          "bsonType": "int"
+                        },
+                        "embedded": {
+                            "title": "FlexEmbeddedObject",
+                            "bsonType": "object",
+                            "required": [
+                                "embeddedName"
+                            ],
+                            "properties": {
+                                "embeddedName": {
+                                    "bsonType": "string"
+                                }
+                            }
+                        }
+                    },
+                    "required": [
+                        "name",
+                        "section",
+                        "age",
+                        "_id"
+                    ],
+                    "title": "FlexParentObject"
+                
+                }
+            }
+            """.trimIndent()
+        )
+    }
+
+    @Suppress("LongMethod")
+    suspend fun AppServicesClient.initializePartitionSync(
+        app: BaasApp,
+        service: Service
+    ) {
+        val databaseName = app.clientAppId
```

Comment:
Maybe I lost track of the code, but we should strive to have a database pr. app, but it looks like this is doing that?

Reply:
Oh, this allows each app to have a separate DB. I forgot to mention it on the PR description.